### PR TITLE
feat: support parquet file source

### PR DESCRIPTION
**Pull request by forsaken628**
#2521

### CODE REVIEW

File: internal/io/file/parquet_reader.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+type ParquetReader struct {
+	f          *os.File
+	pf         *parquet.File
+	groups     []parquet.RowGroup
+	curGroup   int
+	rowsReader parquet.Rows
+}
+
+func CreateParquetReader(filename string, config *FileSourceConfig, ctx api.StreamContext) (FormatReader, error) {
```
Comment (ngjaying) [inline internal/io/file/parquet_reader.go:33]: Recommend to put ctx as the first arg

File: internal/io/file/parquet_reader.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	var pr ParquetReader
```
Comment (ngjaying) [inline internal/io/file/parquet_reader.go:34]:
```
var (
  pr ParquetReader
  err error
)
```

**Review by ngjaying — CHANGES_REQUESTED** [review]
Is the parquet mod big? Can you check how big the binary size will increase? Thanks.

**Comment by forsaken628** [comment]
> Is the parquet mod big? Can you check how big the binary size will increase? Thanks.

before
24M kuiper
38M kuiperd
after
27M kuiper
41M kuiperd
There is another option, but I haven't compared it before.
https://pkg.go.dev/github.com/apache/arrow/go/v17/parquet

**Comment by ngjaying** [comment]
> > Is the parquet mod big? Can you check how big the binary size will increase? Thanks.
> 
> linux before 24M kuiper 38M kuiperd after 27M kuiper 41M kuiperd There is another option, but I haven't compared it before. https://pkg.go.dev/github.com/apache/arrow/go/v17/parquet

Great, thanks! If anyone complians about the size, we can make it optional by build tag later.

**Review by ngjaying — APPROVED** [review]

**Pull request merged**